shared/chat/conversation/info-panel/attachments.tsx

```diff
@@ -0,0 +1,577 @@
+        return (
+          <Kb.Button
+            type="Default"
+            mode="Secondary"
+            label="Load more"
+            onClick={onLoadMore}
+            style={styles.loadMore}
+          />
+        )
+      } else if (status === 'loading') {
+        return <Kb.ProgressIndicator style={styles.loadMoreProgress} />
+      } else if (status === 'error') {
+        return (
+          <Kb.Button
+            type="Danger"
+            mode="Secondary"
+            label="Error loading, try again"
+            onClick={onRetry}
+            style={styles.loadMore}
+          />
+        )
+      }
+      return null
+    },
+    renderSectionHeader: () => {
+      return null
+    },
+  }
+}
+
+type Sizing = {
+  dims: {
+    height: number
+    width: number
+  }
+  margins: {
+    marginBottom: number
+    marginLeft: number
+    marginRight: number
+    marginTop: number
+  }
+}
+
+type Thumb = {
+  ctime: number
+  height: number
+  isVideo: boolean
+  onClick: () => void
+  previewURL: string
+  width: number
+}
+
+type MediaThumbProps = {
+  thumb: Thumb
+  sizing: Sizing
+}
+
+type MediaThumbState = {
+  loading: boolean
+}
+
+class MediaThumb extends React.Component<MediaThumbProps, MediaThumbState> {
+  state = {loading: true}
+  _setLoaded = () => {
+    this.setState({loading: false})
+  }
+  render() {
+    const {sizing, thumb} = this.props
+    return (
+      <Kb.Box2 direction="vertical" style={styles.thumbContainer}>
+        <Kb.ClickableBox onClick={thumb.onClick} style={{...sizing.margins}}>
+          <Kb.Image src={thumb.previewURL} style={{...sizing.dims}} onLoad={this._setLoaded} />
+        </Kb.ClickableBox>
+        {!!thumb.isVideo && (
+          <Kb.Box2 direction="vertical" style={styles.durationContainer}>
+            <Kb.Icon type="icon-film-64" style={Kb.iconCastPlatformStyles(styles.filmIcon)} />
+          </Kb.Box2>
+        )}
+        {this.state.loading && <Kb.ProgressIndicator style={styles.loading} />}
+      </Kb.Box2>
+    )
+  }
+}
+
+const rowSize = 4
+
+export class MediaView {
+  _clamp = (thumb, maxThumbSize) => {
+    return thumb.height > thumb.width
+      ? {height: (maxThumbSize * thumb.height) / thumb.width, width: maxThumbSize}
+      : {height: maxThumbSize, width: (maxThumbSize * thumb.width) / thumb.height}
+  }
+  _resize = thumb => {
+    const maxThumbSize = Styles.isMobile ? imgMaxWidthRaw() / rowSize : 80
+    const dims = this._clamp(thumb, maxThumbSize)
+    const marginHeight = dims.height > maxThumbSize ? (dims.height - maxThumbSize) / 2 : 0
+    const marginWidth = dims.width > maxThumbSize ? (dims.width - maxThumbSize) / 2 : 0
+    return {
+      dims,
+      margins: {
+        marginBottom: -marginHeight,
+        marginLeft: -marginWidth,
+        marginRight: -marginWidth,
+        marginTop: -marginHeight,
+      },
+    }
+  }
+
+  _formRows = thumbs => {
+    let row = []
+    return thumbs.reduce((l, t, index) => {
```

Comment:
I think it would be cleaner if we used some lodash utility fns.
i.e.:
```js
_.chunk(thumbs.map(thumb => ({sizing: this._resize(thumb), thumb})), rowSize)
```